includes/process-download.php

```diff
@@ -867,7 +867,7 @@ function edd_process_signed_download_url( $args ) {
 	}
 
 	$order_parts = explode( ':', rawurldecode( $_GET['eddfile'] ) );
-	$price_id    = isset( $order_parts[3] ) ? (int) $order_parts[3] : null;
+	$price_id    = edd_has_variable_prices( $order_parts[1] ) && isset( $order_parts[3] ) ? (int) $order_parts[3] : null;
```

Comment:
```suggestion
	$price_id    = isset( $order_parts[3] ) && edd_has_variable_prices( $order_parts[1] ) ? (int) $order_parts[3] : null;
```
I think if we change the order then the edd function would only be called if the `isset()` is true, right? And might save a microsecond somewhere?

Reply:
@robincornett I don't believe it will make a difference, as it is always there and sets to `0` as a default for `null` (the old issue showing it's head). But yeah we can reverse it.

Reply:
I think it could make a difference, as the `edd_get_download_file_url` function now will not set the fourth order part if the download is not a variable product and the price ID is properly set to `null` (which it will be for new and migrated orders) -- that was changed in #9179. Tiny but feels worth doing.